phys/module_irrigation.F

```diff
@@ -1,6 +1,46 @@
 MODULE module_irrigation
 CONTAINS
 
+SUBROUTINE irrigation_on(julian_in,irr_start_julianday,irr_end_julianday,irr_on,xtime,gmt,irr_start_hour,end_hour)
+ IMPLICIT NONE
+ REAL, INTENT(IN) :: julian_in, xtime, gmt
+ INTEGER, INTENT(IN) :: irr_start_julianday, irr_end_julianday, irr_start_hour, end_hour
+ INTEGER :: irr_on_time, irr_on_day, tloc, xt24
+ INTEGER, INTENT(OUT):: irr_on
+
+ irr_on_day = 0
+ irr_on_time = 0
+ irr_on = 0
+
+ xt24=mod(xtime,1440.)
+ tloc=floor(gmt+xt24/60.)
+ if(tloc.lt.0) tloc=tloc+24
+ 
+ IF(irr_start_julianday .lt. irr_end_julianday) THEN
+   IF((julian_in.GE.irr_start_julianday .AND. julian_in.LT.irr_end_julianday))  THEN
+      irr_on_day=1
+   END IF
+ ELSE
+   IF((julian_in.GE.irr_start_julianday  .AND.  julian_in.LT.367) .OR.  &
+     (  julian_in.GE.0 .AND.   julian_in.LT.irr_end_julianday))  THEN
+      irr_on_day = 1
+   END IF
+ END IF 
+ 
+ IF(irr_start_hour .lt. end_hour) THEN
+   IF((tloc.GE.irr_start_hour .AND. tloc.LT.end_hour))  THEN
+      irr_on_time=1
+   END IF
+ ELSE
+   IF((tloc.GE.irr_start_hour  .AND.  tloc.LT.24) .OR.  &
+     (  tloc.GE.0 .AND.   tloc.LT.end_hour))  THEN
```

Comment:
This could have been simpler logic if it was just .not. between start and end.

Reply:
yes. I change it now